daemon/daemon.go

```diff
@@ -380,8 +380,22 @@ func (daemon *Daemon) restore() error {
 					}
 				}
 			}
+
 			if err := daemon.containerStart(c); err != nil {
```

Comment:
Right now libnetwork only returns a `fmt.Errorf` error, it would be nice to have a typed error that we could check instead of assuming a discovery issue